src/geometry/manifold/ManifoldGeometry.cc

```diff
@@ -76,17 +78,31 @@ std::string ManifoldGeometry::dump() const {
 }
 
 std::shared_ptr<const PolySet> ManifoldGeometry::toPolySet() const {
-  auto ps = std::make_shared<PolySet>(3);
   manifold::Mesh mesh = getManifold().GetMesh();
-  ps->reserve(mesh.triVerts.size());
-  Polygon poly(3);
-  for (const auto &tv : mesh.triVerts) {
-    for (const int j : {0, 1, 2}) {
-      poly[j] = vector_convert<Vector3d>(mesh.vertPos[tv[j]]);
-    }
-    ps->append_poly(poly);
+  std::vector<int> indmap;
+  indmap.reserve(mesh.vertPos.size());
+  PolySetBuilder builder(mesh.vertPos.size(),mesh.triVerts.size());
+  for(const auto &pt :mesh.vertPos)  
+    indmap.push_back(builder.vertexIndex({pt[0],pt[1],pt[2]}));
+  for (const auto &tv : mesh.triVerts) 
+      builder.appendPoly({indmap[tv[0]],indmap[tv[1]],indmap[tv[2]]});
+  return std::shared_ptr<PolySet>(builder.build());
+/*
+  PolySet *ps = new PolySet(3);
+  ps->vertices.reserve(mesh.vertPos.size());
+  for(int i=0;i<mesh.vertPos.size();i++) {
+	auto pt=mesh.vertPos[i];
+  	ps->vertices.push_back(Vector3d(pt[0],pt[1],pt[2]));
   }
-  return ps;
+  ps->indices.reserve(mesh.triVerts.size());
+  for(int i=0;i<mesh.triVerts.size();i++) {
+	auto tri=mesh.triVerts[i];
+	std::vector<int> trinew({tri[0],tri[1],tri[2]});
+  	ps->indices.push_back(trinew);
+  }
+
+  return std::shared_ptr<PolySet>(ps);
+*/
```

Comment:
what is this commented code?

Reply:
Could you add exactly that as a code comment, or remove it and add it to #4851 ?